### PR TITLE
Nova chance: Enable user merging when accepting team invites

### PR DESCRIPTION
**Pull request by GregWeil**
Stop suppressing the authorization header in the api request. Thanks to some api changes made by David, accepting invites sent to email addresses is a bit more predictable. It'll now only merge if the requesting user is anonymous, and will leave you signed in as the invited user afterwards (so it'll work like an email logon).

Sending team invites to email addresses is still behind a feature toggle, do you think I should go ahead and enable it in this pr?

@davidhodgson

### CODE REVIEW

**Comment by davidhodgson** [comment]
A very similar route to joining a team was the email sign in, e.g.: https://glitch.com/login/email?token={token}. Were we suppressing the authorization header there as well? (I'm guessing not).

**Comment by GregWeil** [comment]
No, we only suppress the header for team invites

**Comment by ttseng** [comment]
A few things I tested with some unexpected results:

Case 1: Clicking join team link via email for an anonymous user
**Steps:**
1. Created a project as an anonymous account
2. In a new browser window, under my own account (@scientiffic), I sent a team invite to a "wrdstomakeup@yahoo.com" (a test email account)
3. I copied the link from the email and opened it in the same window I was active as an anonymous user.
4. A new user seemed to have been created for that particular email address; this user is distinct from the anonymous user I am currently logged in as:

screen shot 2019-01-24 at 5 46 11 pm

**Actual Result:**
A new user account is created based on the email address the team invite was sent to, where the new user handle matches that of the email address

**Expected Result:**
Should the anonymous user be merged with the invited user such that the email address wrdstomakeup@yahoo.com is now associated with the anonymous user?

Case 2: Clicking join team link via email for an existing user
**Steps:**
1. Opened Glitch as an anonymous user.
2. In a new browser window, under my own Glitch account (@scientiffic), I sent a team invite to "ttseng@alum.mit.edu," an email address associated with the Glitch user @scientiffic-test by way of Github
screen shot 2019-01-24 at 5 44 41 pm

**Actual Result:** 
I ended up seeing the option to invite the existing user ttseng@alum.mit.edu (whose avatar does not match @scientiffic-test) - as well as an option to invite anyone with an @alum.mit.edu email.

**Expected Result:**
Did not expect to see a whitelist option.  Expected the user avatar to match the one for @scientiffic-test because @scientiffic-test 

3. Then I clicked the Join Team link from the email in the browser where I was working as an anonymous user.  At this point, I see that the existing user @scientiffic-test was added to the team.

screen shot 2019-01-24 at 5 57 28 pm

**Actual Result**:
The anonymous user seems to have accepted the invite to join the team, on behalf of an existing user

**Expected Result:**
The join team link should only be valid if the logged in user matches the invited user.

Let me know if any of this doesn't make sense - I can try to capture the entire workflow in a video.

**Comment by GregWeil** [comment]
Status update: does not merge for Tiff, does merge correctly for me. It doesn't seem browser dependent, so I'll have to do some more digging.

**Comment by GregWeil** [comment]
Status update again! user merging is working but the api returns a 500 so you get logged out

File: src/presenters/includes/user-result-item.js [hunk 1]
```diff
@@ -64,7 +64,8 @@ export class InviteByEmail extends React.Component {
       <button onClick={this.props.onClick} className="button-unstyled result">
         <img className="avatar" src={ANON_AVATAR_URL} style={style} alt="" />
         <div className="result-name">
-Invite
+          Invite
+          {' '}
           {this.props.email}
```
Comment (GregWeil) [inline src/presenters/includes/user-result-item.js]: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-one-expression-per-line.md

This rule seems kind of intrusive, I'd consider all on one line to be more readable here

**Comment by GregWeil** [comment]
@ttseng this is working now, could you review it again?

**Comment by ttseng** [comment]
just tried and I wasn't able to merge the user account as expected.  here's what I did

1. enabled email login on 'nova-chance' at '/secret'
2. created a session with an anonymous user and created a hello world project for this anonymous user.
3. invited 'tiff+1@glitch.com' to a team using my personal Glitch account 'scientiffic'
3. opened up the invitation code from the email at 'nova-chance' while "logged"" in as the anonymous user.  expected that a new user would have been created, with the name matching the email but still with the anonymous user's hello world project.  the actual result was that it created a new user with a name matching the username, but the new user doesn't have the hello world project.

I can demo for you tomorrow if it's helpful.

**Comment by ttseng** [comment]
working for me - opened invite link on https instead of http in previous attempt

**Pull request merged**